Library/ENV/4.3/cc

```diff
@@ -1,6 +1,10 @@
-#!/System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/ruby -W0
+#!/bin/sh
+# Make sure this shim uses the same Ruby interpreter that is used by Homebrew.
+unset RUBYLIB
+unset RUBYOPT
+exec "$HOMEBREW_RUBY_PATH" -x "$0" "$@"
```

Comment:
Thanks for the explanation. One last thing: should we add below sanity check?
```bash
[[ -n "$HOMEBREW_RUBY_PATH" ]] || echo "The build-tool has reset ENV. --env=std required." >&2
```

Notice that we have similar check in the ruby code https://github.com/Homebrew/homebrew/blob/master/Library/ENV/4.3/cc#L307